www/tests/test_app.py

```diff
@@ -1,4 +1,11 @@
 import app
 
-def test_dummy():
+def test_sensor_init():
     assert len(app.binary_sensors) > 0
+
+def test_get_local_ip():
```

Comment:
I think you'll need a way to exit the loop. A `shutdown` variable or something like it that it's checking every time through the loop. If it were part of a class, you could have a class method to start and stop the thread. I'm not sure if the socketio background task will accept a class member or not.